src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -761,35 +757,39 @@ public struct RequestEventDispatcher
     /***************************************************************************
 
         Returns:
-            true if any registered fibers wish to yield
+            flags to pass to EventDispatcher.nextEvent(), with Resume and/or
+            Yield bits set, based on the events that registered fibers are
+            waiting for.
 
     ***************************************************************************/
 
-    private bool waitingYielders ( )
+    RequestOnConnBase.EventDispatcher.NextEventFlags nextEventYieldResumeFlags ()
     {
-        foreach ( waiting_fiber; this.waiting_fibers.array() )
-            if ( waiting_fiber.enabled &&
-                waiting_fiber.event.active == waiting_fiber.event.active.yield )
-                return true;
-
-        return false;
-    }
+        RequestOnConnBase.EventDispatcher.NextEventFlags flags;
 
-    /***************************************************************************
-
-        Returns:
-            true if any registered fiber is awating a signal.
-
-    ***************************************************************************/
-
-    private bool waitingForSignals ( )
-    {
         foreach ( waiting_fiber; this.waiting_fibers.array() )
-            if ( waiting_fiber.enabled &&
-                waiting_fiber.event.active == waiting_fiber.event.active.signal )
-                return true;
+        {
+            if ( waiting_fiber.enabled )
+            {
+                switch (waiting_fiber.event.active)
+                {
+                    case waiting_fiber.event.active.signal:
+                        flags |= flags.Resume;
+                        if (flags & ~cast(int)(flags.Resume))
```

Comment:
Hmm... wouldn't `flags.Receive` be required to also be set for this condition to be true?

Reply:
This just does early return if the `Yield` flag was already set, so we can stop the iteration. Unless I'm missing something? I'm just comparing to non-zero of the resulting bitwise and.

So, if there was a yield set previously:

```
[resume][yield][receive]
0       1       0
```

and-ing this with `011` (`~resume`) will yield `!= 0` so we can return immediately  .

Reply:
Ahh gotcha. Would `if ( flags & flags.Yield )` be clearer and equivalent? (As we're only talking about yield / resume.)

Reply:
Yeah, that's the way I originally wrote this, but this is the more friendly when it comes to refactoring and adding the next code (in each case you just repeat the flag you're setting).

Reply:
Oh, no, that's wrong!

Reply:
Hmmm... it's tricky though. You only want to return if all flags _that this function deals with_ are set. (i.e. not `Receive`.)

Reply:
Updated.